iwyu_path_util.cc

```diff
@@ -82,9 +82,11 @@ string Basename(const string& path) {
 }
 
 string GetCanonicalName(string file_path) {
-  // Get rid of any <> and "" in case file_path is really an #include line.
-  StripLeft(&file_path, "\"") || StripLeft(&file_path, "<");
-  StripRight(&file_path, "\"") || StripRight(&file_path, ">");
+  // For this special 'path' we just remove the quotes.
+  if (file_path == "<built-in>")
+    return "built-in";
```

Comment:
Sure? This keeps the same behavior as before (removing the `<>`) and I don't know what happens for the `BaseName("<built-in>")` stuff, although I think it will just return it unmodified and we are better of using `<built-in>` to distinguish it from an include named "build-in".